quinn-proto/src/shared.rs

```diff
@@ -134,6 +122,52 @@ impl fmt::Display for ConnectionId {
     }
 }
 
+/// Generates connection IDs for incoming connections
+pub trait ConnectionIdGenerator: Send {
```

Comment:
Is the `shared` module actually `pub`lically exported? Otherwise won't be usable 😀
Btw: It could also live an a seperate `connection_id_generator` module, or just be reexported from the top level module.

Reply:
Will do. This sounds good as the customized cid generator implementation can be done in a different crate and later imported into quinn-proto.